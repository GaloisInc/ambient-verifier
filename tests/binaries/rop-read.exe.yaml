statechart:
  name: ROPViaRead
  root state:
    name: Root
    initial: NormalExecution
    states:
      - name: NormalExecution
        description: |
          Execution before the Weird Machine starts
        transitions:
          - target: WMEntered
            event: EnterWeirdMachine(0x401000)
      - name: WMEntered
        description: |
          The Weird Machine has been entered by clobbering the return address
        transitions:
          - target: DidExec
            event: IssuesExecveSyscall
      - name: DidExec
        description: |
          The ROP attack has completed
        type: final
