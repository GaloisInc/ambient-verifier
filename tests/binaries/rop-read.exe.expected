successful: 0
failed: 0
wmTargets: [0x401000]
fsRoot: tests/filesystems/rop-read
